### ISSUE TITLE
Setting to sort JSON keys when formatting JSON with the built-in language service

### ISSUE THREAD

**Issue by earshinov** [microsoft/vscode#152480]
Hey,

I would like to request a new setting like "json.format.sort_keys" (false by default).  When set to true, it should make the formatter built into VS Code (triggered by the "Format Document" command in a file with Language Mode set to JSON) sort JSON keys alphabetically.

#### Illustration

Original JSON:

```json
{"b":2,"a":1}
```

After "Format Document", current behavior (*subject to indentation preferences):

```json
{
    "b": 2,
    "a": 1
}
```

Proposed behavior with `"json.format.sort_keys": true`:

```json
{
    "a": 2,
    "b": 1
}
```

#### Additional considerations

- Keys of nested objects should be sorted, too.

#### Use case

I sometimes use JSON formatting to normalize JSON configurations obtained from several installations of a particular software, in order to `diff` them. It would tremendously help if keys were sorted right away. Now I have to use an external tool for that.

#### References:

Version information:
- VS Code 1.68.0

My `settings.json`:

```json
{
  "[json]": {
    "editor.defaultFormatter": "vscode.json-language-features"
  }
}
```

**Comment by earshinov** [comment]
In the use case I gave as an example, key order is arbitrary in the original files. It makes diffs unreadable. When keys are sorted, the structure of the file is consistent and it is easy to see which keys were added, which were removed and which had respective values modified.

Personally I never needed to sort properties by value. Also, once VS Code allows to sort properties by keys OR values, I assume people would want to sort some objects by key and other objects by value etc. In my opinion, this will make things unnecessary complicated. If you ask me, I would advise to start with sorting by keys only. Other kinds of sorting can be added later if needed.

**Comment by ricahardHaggioGwati** [comment]
Well, I see, in that particular use case, I can see the advantage of sorting keys.  Once sorting keys has been implemented then it's a matter of time before more devs want to sort both. I think that the best course of action for a single feature would be the sorting of both keys and their respective values combined as an individual value.

**Comment by earshinov** [comment]
Honestly, I don't see a point of combining keys with values because normally (in 99.999% cases, I presume) one doesn't have duplicate keys.

**Comment by ricahardHaggioGwati** [comment]
They might be related information from the key to its value, for instance in a recent project that I have been working on they is an error message key that contains its message as a value. If I wanted to sort that JSON data, It would not be helpful to split that key from its value

**Comment by aeschli** [comment]
I assume the result should be 
```
{
    "a": 1,
    "b": 2
}
```

**Comment by earshinov** [comment]
@aeschli , Yes, of course, thank you for noticing. I edited the description.

**Comment by aeschli** [comment]
IMO a formatter should only add or remove whitespaces (space, new lines), and not remove, add or reorder tokens.
But the requested functionality makes sense and can be provided by a separate command.

There's already an extension that does it: https://marketplace.visualstudio.com/items?itemName=richie5um2.vscode-sort-json

Given that I don't think this should be part of the formatter, I close the issue as `won't fix`
